demisto_sdk/commands/download/downloader.py

```diff
@@ -748,520 +985,487 @@ def get_playbook_id_by_playbook_name(self, playbook_name: str) -> Optional[str]:
-                self.files_not_downloaded.append([file_name, reason])
-            elif file_type == "script":
-                reason = "Downloading a script written in JavaScript is not supported."
-                self.files_not_downloaded.append([file_name, reason])
-            return False
-        return True
+        Download the files after processing is done to the output directory.
+        For integrations / scripts, YAML extraction is done.
+        Content items that already exist in the output pack, will be skipped, unless the '--force' flag is used.
+        If it is, the existing and downloaded YAML files will be merged, as some fields are deleted by the server.
 
-    def merge_into_pack(self) -> None:
-        """
-        Merges the custom content into the output pack.
-        For integrations/scripts an extraction will be made and the relevant files will be merged into the pack.
-        If force flag is given then the function will add new files and "Smartly" merge existing files such that
-        important fields deleted by Demisto will be kept. If no force is present, it will only download new files.
+        Args:
+            downloaded_content_objects (dict[str, dict]): A dictionary of content objects to download.
+            existing_pack_structure (dict[str, list]): A dictionary of existing content objects in the output path.
+            output_path (Path): The output path to write the files to.
+
+        Returns:
+            bool: True if all files were downloaded successfully, False otherwise.
         """
-        for custom_content_object in self.custom_content:
-
-            file_entity: str = custom_content_object[
-                "entity"
-            ]  # For example: Integrations
-            exist_in_pack: bool = custom_content_object[
-                "exist_in_pack"
-            ]  # For example: True
-            file_name: str = custom_content_object["name"]  # For example: Hello World
-            file_ending: str = custom_content_object["file_ending"]  # For example: yml
-
-            if exist_in_pack:
-                if self.force:
-                    if file_entity in (INTEGRATIONS_DIR, SCRIPTS_DIR):
-                        self.merge_and_extract_existing_file(custom_content_object)
-                    else:
-                        self.merge_existing_file(custom_content_object, file_ending)
+        logger.info("Saving downloaded files to output path...")
+        successful_downloads_count = 0
+        existing_files_skipped_count = 0
+        failed_downloads_count = 0
+
+        for file_name, content_object in downloaded_content_objects.items():
+            content_item_name: str = content_object["name"]
+            content_item_entity: str = content_object["entity"]
+            content_item_skipped = False
+
+            try:
+                if content_item_entity in (INTEGRATIONS_DIR, SCRIPTS_DIR):
+                    file_downloaded = self.download_unified_content(
```

Comment:
They can't be merged into a single one (without making a big mess, I've tried), since for unified items we first download the unified file to disk (temp directory), split it, and then parse the newly generated split files, and check for each unsplitted file if it already exists, while for non-unified we first check if it exists (since we already know the final filename), and only then download it if relevant. So the logic is somewhat reversed for when we download the content item.

I will, however, try to move some of the code that is shared between them into a separate method.